src/config-schemas/vitals-config-schema.ts

```diff
@@ -0,0 +1,83 @@
+import { validators } from "@openmrs/esm-module-config";
+
+export const vitalsConfigSchema = {
+  vitalsEncounterTypes: {
+    arrayElements: {
+      uuid: { validators: [validators.isString] },
```

Comment:
:grimacing: Do we *have to* use UUIDs? Is it possible to use something a little more human-friendly?

Reply:
:laughing: I couldn't think of another way to do it that would work with REST. What do you propose?

Reply:
I mean, I think we should use names whenever possible. FSN for concepts. Encounter name for encounters. etc. But yeah, I suppose we might be constrained by the API. @ibacher , do you have thoughts about this?

Reply:
`EncounterType`s always have a name which should be unique. You should be able to get it from REST like this: `/ws/rest/v1/encountertype?q=<name>`.

Reply:
I have concerns with using this pattern for configurations i.e I would try to minimize hitting the REST endpoint unnecessarily to resolve configs. For the above scenarios, I would need to make several api calls to resolve to uuids. To me, readability of configs feels like a design/set-up time problem, and wouldn't want to tax the app later on. We actually ran into performance issues with our angular client due to many REST api calls, and had to optimize by eliminating unnecessary calls (worked perfectly on the dev environment, until it was deployed and you have users with less powerful tablets trying to use the app on a low-bandwidth 3g internet. The UX was terrible). We could try to solve the readability problem by either 1)including fields that will describe a config or 2) building tools to improve the overall experience of creating/viewing the config files.

Reply:
Oh yeah, I agree, we shouldn't make *extra* calls to avoid UUIDs in the configs. It might be worth checking whether the visit data can be requested so that it comes with the encounter type name. If not though, no worries, it sounds like you're weighing this well.